Delcam.ProductInterface.PowerMILL/Classes/PMBoundariesCollection.cs

```diff
@@ -184,6 +184,14 @@ public PMBoundary CreateBoundary(Spline spline)
             return newBoundary;
         }
 
+        /// <summary>
+        /// Undraw all boundarys
+        /// </summary>        
+        public void UndrawAll()
+        {
+            _powerMILL.DoCommand("UNDRAW BOUNDARY ALL");
```

Comment:
I think you could simplify all of these by putting the function on the base class PMEntity and use the code:
`_powerMILL.DoCommand("UNDRAW` " + Identifier + " ALL");

Reply:
Sorry for my long abstinence, but there was too much regular work to do.

You're right, that's an much cleaner and easier way to do this. But if i'm adding the function to the PMEntity class, the function would only be accessible from an specific entity.

e.g.
```
PMToolpath TPTest;   
TPTest.Undraw;
```

This is fine and i will add this function later, but at the moment i'm at the Undraw**ALL** function. So i think i must insert that function in the PMEntitiesCollection class. But here i encountered one problem. There is no "Idenifier" at the PMEntitiesCollection class. The only "workaround" i've found is this:

```
public void UndrawAll()
 {             
     if (Count > 0)
     {                
         _powerMILL.DoCommand("UNDRAW " + this[0].Identifier + " ALL");
     }
 }
```
Is there another, easier way to get the identifier?

Reply:
I believe you should be able to use:
`T.Identifier`
It is a generic class and T gives you access to the actual type at runtime.

Reply:
I tried it but i'm getting this error:
```
CS0119: 'T' is a 'type parameter', which is not valid in this context"
```

Reply:
Hmmm, apparently that's not allowed:
https://stackoverflow.com/questions/196661/calling-a-static-method-on-a-generic-type-parameter
There is some discussion in the above post that you could either use reflection or else use:
`new T().Identifer`

Reply:
OK, this is uncharted territories for me. Will try to figure this out. Thanks for the hint